app/packages/backend/src/index.ts

```diff
@@ -1,130 +1,100 @@
+      }),
+    ],
+  }),
+);
 
-  const apiRouter = Router();
-  apiRouter.use(currentRequestMiddleware.handler);
-  apiRouter.use('/catalog', await catalog(createEnv('catalog')));
-  apiRouter.use('/scaffolder', await scaffolder(createEnv('scaffolder')));
-  apiRouter.use('/auth', await auth(createEnv('auth')));
-  apiRouter.use('/techdocs', await techdocs(createEnv('techdocs')));
-  apiRouter.use('/proxy', await proxy(createEnv('proxy')));
-  apiRouter.use('/search', await search(createEnv('search')));
-  apiRouter.use('/azure-devops', await azureDevOps(createEnv('azure-devops')));
-  apiRouter.use('/kubernetes', await kubernetes(createEnv('kubernetes')));
-  apiRouter.use('/permission', await permission(createEnv('permission')));
-  apiRouter.use('/adp', await adp(createEnv('adp')));
+// AuthN and AuthZ
+backend.add(import('@backstage/plugin-auth-backend'));
+backend.add(import('@backstage/plugin-auth-backend-module-microsoft-provider'));
+backend.add(import('@backstage/plugin-permission-backend/alpha'));
+backend.add(addAdpPermissionsPolicy);
+backend.add(addCatalogPermissionRules);
 
-  // Add backends ABOVE this line; this 404 handler is the catch-all fallback
-  apiRouter.use(notFoundHandler());
+// Backstage
+backend.add(import('@backstage/plugin-app-backend/alpha'));
+backend.add(import('@backstage/plugin-catalog-backend/alpha'));
+backend.add(
+  import('@backstage/plugin-catalog-backend-module-scaffolder-entity-model'),
+);
+backend.add(import('@backstage/plugin-catalog-backend-module-github/alpha'));
+backend.add(import('@backstage/plugin-catalog-backend-module-msgraph/alpha'));
+backend.add(addAdoNameTransformer);
+backend.add(addAdpDatabaseEntityProvider);
+backend.add(import('@backstage/plugin-scaffolder-backend/alpha'));
+backend.add(import('@backstage/plugin-scaffolder-backend-module-github'));
+backend.add(addScaffolderModuleAdpActions);
+backend.add(
+  import('@roadiehq/scaffolder-backend-module-http-request/new-backend'),
+);
+backend.add(import('@backstage/plugin-search-backend/alpha'));
+backend.add(import('@backstage/plugin-search-backend-module-catalog/alpha'));
+backend.add(import('@backstage/plugin-search-backend-module-techdocs/alpha'));
+backend.add(import('@backstage/plugin-techdocs-backend/alpha'));
+backend.add(import('@backstage/plugin-kubernetes-backend/alpha'));
+backend.add(import('@backstage/plugin-proxy-backend/alpha'));
+backend.add(import('@backstage/plugin-azure-devops-backend'));
 
-  const service = createServiceBuilder(module)
-    .loadConfig(config)
-    .addRouter('/api', apiRouter)
-    .addRouter('', await app(createEnv('app')));
+// ADP
+backend.add(legacyPlugin('adp', import('./plugins/adp')));
 
-  await service.start().catch(err => {
-    console.log(err);
-    process.exit(1);
-  });
-}
-
-module.hot?.accept();
-main().catch(error => {
-  console.error('Backend failed to start up', error);
-  process.exit(1);
-});
+backend.start();
```

Comment:
unawaited promise, add a .catch to this, or `void backend.start()` to make it clear that we dont care if it fails.